openghg/store/_boundary_conditions.py

```diff
@@ -160,109 +187,57 @@ def read_file(
-            metadata["min_longitude"] = round(float(bc_data["lon"].min()), 5)
-            metadata["max_latitude"] = round(float(bc_data["lat"].max()), 5)
-            metadata["min_latitude"] = round(float(bc_data["lat"].min()), 5)
-            metadata["min_height"] = round(float(bc_data["height"].min()), 5)
-            metadata["max_height"] = round(float(bc_data["height"].max()), 5)
-
-            metadata["input_filename"] = filepath.name
-
-            metadata["time_period"] = period_str
 
-            key = "_".join((species, bc_input, domain))
+        # Check to ensure no required keys are being passed through optional_metadata dict
+        self.check_info_keys(optional_metadata)
+        if optional_metadata is not None:
+            additional_metadata.update(optional_metadata)
 
-            boundary_conditions_data: dict[str, dict] = {}
-            boundary_conditions_data[key] = {}
-            boundary_conditions_data[key]["data"] = bc_data
-            boundary_conditions_data[key]["metadata"] = metadata
-
-            matched_keys = set(metadata) & set(fn_input_parameters)
-            additional_input_parameters = {
-                key: value for key, value in fn_input_parameters.items() if key not in matched_keys
-            }
-
-            # Check to ensure no required keys are being passed through optional_metadata dict
-            self.check_info_keys(optional_metadata)
-
-            if optional_metadata is not None:
-                additional_metadata.update(optional_metadata)
+        # Mop up and add additional keys to metadata which weren't passed to the parser
+        boundary_conditions_data = self.update_metadata(
+            boundary_conditions_data, additional_input_parameters, additional_metadata
+        )
 
-            # Mop up and add additional keys to metadata which weren't passed to the parser
-            boundary_conditions_data = self.update_metadata(
-                boundary_conditions_data, additional_input_parameters, additional_metadata
-            )
+        # This performs the lookup and assignment of data to new or
+        # existing Datasources
+        data_type = "boundary_conditions"
+        datasource_uuids = self.assign_data(
+            data=boundary_conditions_data,
+            if_exists=if_exists,
+            new_version=new_version,
+            data_type=data_type,
+            compressor=compressor,
+            filters=filters,
+        )
 
-            # This performs the lookup and assignment of data to new or
-            # existing Datasources
-            datasource_uuids = self.assign_data(
-                data=boundary_conditions_data,
-                if_exists=if_exists,
-                new_version=new_version,
-                data_type=data_type,
-                compressor=compressor,
-                filters=filters,
-            )
+        # TODO: MAY NEED TO ADD BACK IN OR CAN DELETE
```

Comment:
I know this was just flagged because it was reformatted, but can this commented out code be removed?

Reply:
Yes, this commented out code is still hanging around in a few places - this functionality was to make sure certain keys were updated in the internal metadata for each Datasource (start_date, end_date and latest_version) but this moved somewhere else when it got generalised to updating _all_ internal metadata from the metastore. I don't think we'd go back to this old way of doing it so we can start removing this when we see it.